core/controllers/suggestion.py

```diff
@@ -174,3 +175,24 @@ def get(self):
 
         self.values.update({'suggestions': [s.to_dict() for s in suggestions]})
         self.render_json(self.values)
+
+
+class EditSuggestionHandler(base.BaseHandler):
+    # @acl_decorators.get_decorator_for_accepting_suggestion(
+    #     acl_decorators.can_edit_suggestion)
+    # TODO regarding permissions.
+    def put(self, suggestion_id):
+        """Edits the submitted suggestion.
+
+        Args:
+            suggestion_id: str. id of suggestion to be edit.
+        """
+        suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+        new_suggestion_content = self.payload.get('change')
+
+        old_suggestion_content = suggestion.change.to_dict()
+        old_suggestion_content['new_value'] = new_suggestion_content[
+            'new_value']
+        suggestion.change = old_suggestion_content
+        suggestion.change = exp_domain.ExplorationChange(suggestion.change)
```

Comment:
This would work only for exploration related suggestions. Ideally this feature should be applicable across all suggestion types (even if the frontend for it isn't implemented yet). So here is what I would suggest:
- Extract the type of the change from the existing suggestion.change object. 
- Create an object of the above type with the new dict
- change suggestion.change to the new object. 

Also, all of this should happen in the service function as mentioned above.